test/unit/org/apache/cassandra/batchlog/BatchlogEndpointFilterTest.java

```diff
@@ -91,7 +142,7 @@ public void shouldSelectLastHostsFromLastNonLocalRacks() throws UnknownHostExcep
                                                          .put("3", InetAddressAndPort.getByName("33"))
                                                          .build();
         
-        Collection<InetAddressAndPort> result = filterBatchlogEndpoints(false, endpoints);
+        Collection<InetAddressAndPort> result = filterBatchlogEndpointsRandom(false, endpoints);
         assertThat(result.size()).isEqualTo(2);
 
         // result should be the last replicas of the last two racks
```

Comment:
does this test actually verify that the last host is selected when `filterBatchlogEndpointsRandom` function is called ? Based on the local modification of the function, yes, it will return the last. But is this what you want to test ?